restapi/datasource_api_object.go

```diff
@@ -62,35 +67,46 @@ func dataSourceRestApiRead(d *schema.ResourceData, meta interface{}) error {
     path,
     path,
     path,
-    d.Id(),
+    "0", // We temporarily set ID of the new object to 0
```

Comment:
Why?

Reply:
I suspected that it was related to that, so appreciate the explanation. I do know that later on, in a success case, `d.SetId` and `obj.id` are set to the correct value before returning, so I agree we're good.

It may just be "safer" to only set it to `"0"` if `d.ID()` returns `""`. What do you think about that? It'll remove any chances of strange things happening.